iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/DanglingLineData.java

```diff
@@ -47,23 +43,20 @@ public DanglingLineData(DanglingLine danglingLine, boolean splitShuntAdmittance)
             return;
         }
 
-        double g1 = splitShuntAdmittance ? danglingLine.getG() * 0.5 : danglingLine.getG();
-        double b1 = splitShuntAdmittance ? danglingLine.getB() * 0.5 : danglingLine.getB();
-        double g2 = splitShuntAdmittance ? danglingLine.getG() * 0.5 : 0.0;
-        double b2 = splitShuntAdmittance ? danglingLine.getB() * 0.5 : 0.0;
-
         Complex v1 = ComplexUtils.polar2Complex(u1, theta1);
 
+        // DanglingLine model has shunt admittance on network side only, so it is not split between both sides.
         Complex vBoundaryBus = new Complex(Double.NaN, Double.NaN);
         if (danglingLine.getP0() == 0.0 && danglingLine.getQ0() == 0.0) {
-            LinkData.BranchAdmittanceMatrix adm = LinkData.calculateBranchAdmittance(danglingLine.getR(), danglingLine.getX(), 1.0, 0.0, 1.0, 0.0, new Complex(g1, b1), new Complex(g2, b2));
+            LinkData.BranchAdmittanceMatrix adm = LinkData.calculateBranchAdmittance(
+                    danglingLine.getR(), danglingLine.getX(), 1.0, 0.0, 1.0, 0.0, new Complex(danglingLine.getG(), danglingLine.getB()), new Complex(0, 0));
             vBoundaryBus = adm.y21().multiply(v1).negate().divide(adm.y22());
         } else {
 
             // Two buses Loadflow
             Complex sBoundary = new Complex(-danglingLine.getP0(), -danglingLine.getQ0());
```

Comment:
This code has an issue where is considers only the DanglingLine load part and ignores the optional generation part. This would need to be fixed in another PR.